nbgrader/server_extensions/assignment_list/handlers.py

```diff
@@ -228,7 +228,10 @@ class CourseListHandler(BaseAssignmentHandler):
 
     @web.authenticated
     def get(self):
-        self.finish(json.dumps(self.manager.list_courses()))
+        courses_list = self.manager.list_courses()
+        if 'value' in courses_list:
```

Comment:
This should check the value of the "success" key. If it's false then the value will be a traceback rather than a list.

Reply:
ok, added the check as `if courses_list['success'] and 'value' in courses_list:`